common/base/Init.cpp

```diff
@@ -360,19 +359,27 @@ void Daemonise() {
   }
 
   // close all fds
-  if (rl.rlim_max == RLIM_INFINITY)
-    rl.rlim_max = 1024;
-  for (i = 0; i < rl.rlim_max; i++)
-    close(i);
+  int maxfd = sysconf(_SC_OPEN_MAX);
+  if (maxfd == -1) {
+    if (rl.rlim_max == RLIM_INFINITY) {
+      maxfd = 1024;
```

Comment:
Correct. 

For our use case, we only ever have 3 fds open at this point anyway.